### PR TITLE
[shell] Fix the Linux configuration end-to-end so the .ini, C++ APIs,…

### PR DESCRIPTION
**Pull request by yufengwangca**
Problem
When using the device config command in the chip-shell example, the output does not match what is in the master .ini files from which the configuration is gleaned:

Summary of Changes
Fix the Linux configuration end-to-end so the .ini, C++ APIs, and shell output are all consistant.

fixes #<1603>

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by turon — APPROVED** [review]
Lgtm

**Review by mspang — APPROVED** [review]

File: src/platform/tests/TestConfigurationMgr.cpp [hunk 1]
```diff
@@ -288,7 +288,7 @@ static void TestConfigurationMgr_PairingCode(nlTestSuite * inSuite, void * inCon
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
 
     NL_TEST_ASSERT(inSuite, pairingCodeLen == strlen(pairingCode));
-    NL_TEST_ASSERT(inSuite, strcmp(buf, "016CB664A86A888D") == 0);
+    NL_TEST_ASSERT(inSuite, strcmp(buf, "16CB664A86A888D") == 0);
```
Comment (turon) [inline src/platform/tests/TestConfigurationMgr.cpp:291]: Why isn't the hex representation a full 8 bytes anymore?

**Review by andy31415 — APPROVED** [review]

File: src/platform/tests/TestConfigurationMgr.cpp (hunk 1 above)
Comment (yufengwangca) [reply src/platform/tests/TestConfigurationMgr.cpp:291]: ./src/include/platform/ConfigurationManager.h:58:        kMaxPairingCodeLength      = 15

The maximum len of pairing code is defined to 15 in both CHIP and Weave. Should it be 16?

Comment (yufengwangca) [reply src/platform/tests/TestConfigurationMgr.cpp]: After checking the logic, I believe we need to set the kMaxPairingCodeLength to 16 to accommodate  hex representation of 8 bytes.

**Review by woody-apple — APPROVED** [review]

**Pull request merged**